### PR TITLE
Fixes #45

### LINKED ISSUE #45: Update question 207
### Thanks for helping!

Thank you for helping to port [this question](../blob/master/questions/207/) from C++17 to C++23.

You'll find the source code in [question.cpp](../blob/master/questions/207/question.cpp), this should normally not need modification.

In C++17, the correct answer is:
> The program is guaranteed to output: `0`

Please verify that this is still correct in C++23. If it is, please do the following:
- [ ] Update [explanation.md](../blob/master/questions/207/explanation.md):
  - [ ] Refer to the correct section numbers
  - [ ] Use updated quotes from those sections (the wording might have changed)
  - [ ] Make sure the rest of the text in the explanation is consistent with the new standard
- [ ] Update [hint.md](../blob/master/questions/207/hint.md) if needed (usually not needed)

*Note that some older questions are missing references to the standard, we don't need to fix that now.*

If you get stuck, please describe the problem in detail in a comment to the issue, and a maintainer will add the `help wanted` tag so others can chime in.

If the correct answer has changed from C++17 to C++23, you can either just leave a comment in this issue, or see the instructions for [updating meta data](../blob/master/METADATA_HOWTO.md).

### PR DESCRIPTION
**Pull request by knatten**
Fixes #45

### CODE REVIEW

File: questions/207/explanation.md [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+>
+> Effects: Equivalent to: `return try_emplace(move(x)).first->second;`
+
+where `try_emplace` is defined by §[map.modifiers]¶4:
+
+> `template <class... Args> pair<iterator, bool> try_emplace(const key_type& k, Args&&... args);`
+>
+> Effects: If the map already contains an element whose key is equivalent to `k`, there is no effect. Otherwise inserts an object of type value_type constructed with `piecewise_construct, forward_as_tuple(k), forward_as_tuple(std::forward<Args>(args)...)`.
+
+`value_type` is just a `typedef` for `pair<const Key, T>`, which in our case is `pair<const int, int>`. So it inserts a `pair(42, int())`.
```
Comment (tocic) [inline questions/207/explanation.md:15]: Maybe it's worth explaining why `pair(piecewise_construct, forward_as_tuple(42), forward_as_tuple(std::forward<Args>(args)...))` equals `pair(42, int())`

File: questions/207/explanation.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
+
+`int()` is a *value-initialization*, which for non-class, non-array types means *zero-initialization* (see §[dcl.init]¶8). And according to §[dcl.init]¶6:
```
Comment (tocic) [inline questions/207/explanation.md:17]:
```suggestion
`int()` is a *value-initialization*, which for non-class, non-array types means *zero-initialization* (see §[dcl.init]¶8). And according to §[dcl.init.general]¶6:
```

File: questions/207/explanation.md (hunk 1 above)
Comment (knatten) [reply questions/207/explanation.md:15]: That would indeed be better. But I'd rather spend time getting the questions ported right now. Feel free to add a commit / PR for that if you want to.

File: questions/207/explanation.md (hunk 2 above)
Comment (knatten) [reply questions/207/explanation.md]: Good catch, fixed

**Review by tocic — APPROVED** [review]

**Pull request merged**